code/workspaces/web-app/src/components/dataStorage/CreateDataStoreForm.js

```diff
@@ -1,15 +1,22 @@
 import { Field, reduxForm } from 'redux-form';
 import PropTypes from 'prop-types';
-import React from 'react';
-import { stackTypes } from 'common';
+import React, { useEffect, useState } from 'react';
 import { renderTextField, renderSelectField, renderTextArea, CreateFormControls } from '../common/form/controls';
 import { syncValidate, asyncValidate } from './newDataStoreFormValidator';
-
-const { DATA_STORE, getStackSelections } = stackTypes;
-const initialValues = { type: 'glusterfs' };
```

Comment:
Yes, for similar reasons - in the web app, the storage config is obtained asynchronously, which I think is too late for reduxForm
```
const CreateDataStoreReduxFrom = reduxForm({
  form: 'createDataStore',
  validate: syncValidate,
  asyncValidate,
  asyncBlurFields: ['name'],
  destroyOnUnmount: false,
  initialValues,
})(CreateDataStoreForm);```

Reply:
You can set the form to reinitialize if the initial values change after the form is mounted. This is what I did in the edit data store dialog to handle the fact that the users that can be added are loaded asynchronously. 
https://github.com/NERC-CEH/datalab/blob/39bfa2ea0c2d06fe64539d55d0960bcd424a8850/code/workspaces/web-app/src/components/dataStorage/editDataStoreForm.js#L62

Thinking about it, I think it would be good to have a default if we can, but if it is going to be a big change then I think it probably isn't worth the extra time.

Reply:
Good idea, updated.